### ISSUE TITLE
Update @types/react-tracking

### ISSUE THREAD

**Issue by Sicria** [nytimes/react-tracking#172]
Is it possible to update the types for the `useTracking` hook to allow for passing in an object.

**Comment by tizmagik** [comment]
Yes, that'd be great. TypeScript types via DefinitelyTyped, [@types/react-tracking](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-tracking), are a community-driven effort currently. Although we do have plans to rewrite the internals to TypeScript and thus export type definitions directly (see #152 for WIP or if you'd like to contribute)

**Comment by trescenzi** [comment]
Hi all. I ran into an issue where, in TS, I couldn't pass a function as the first argument to `useTracking` and then in turn use the return value of that function in `dispatch`. I opened a PR(linked above) and I believe that it's accurate. Not 100% but I've already made the changes in my project's `d.ts` file and everything is working as expected. Let me know if there's anymore information you'd like or if there's more I could do here. 

Thank you! Love the library.

**Comment by trescenzi** [comment]
Just wanted to bump this. I've gotten approval from the definitivelytyped maintainers but will wait till the evening of 4/3, or someone here says otherwise, to merge it in case anyone from this repo has opinions on the change. I've got the changes locally running on a production site so I'm pretty certain it's safe, just want to make sure the owners of this repo don't have any issues with it.

**Comment by tizmagik** [comment]
Those updates look good @trescenzi -- thanks so much for contributing. It's greatly apprecaited!
